metabase/metabase
# [E2E] Deprecate `openNotebookEditor` helper

### Status
PENDING CI

### What does this PR accomplish?
- Deprecates and removes `openNotebookEditor` helper that relied on the UI to reach the "New" data picker
   - Nit: I always found the name a bit misleading because we're literary opening the data picker, not the notebook editor
- Introduces a new helper that encodes a query which leads to the "New > Question" (data picker) page

To test:
- Open a home page `/`
- Click on "New" in the top bar
- Click on "Question"
- Grab the hash from the URL: `eyJjcmVhdGlvblR5cGUiOiJjdXN0b21fcXVlc3Rpb24iLCJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjpudWxsLCJxdWVyeSI6eyJzb3VyY2UtdGFibGUiOm51bGx9LCJ0eXBlIjoicXVlcnkifSwiZGlzcGxheSI6InRhYmxlIiwidmlzdWFsaXphdGlvbl9zZXR0aW5ncyI6e319`
- Decode the hash using [JWT debugger](https://jwt.io/#debugger-io)
- And you should get

```json
{
  "creationType": "custom_question",
  "dataset_query": {
    "database": null,
    "query": {
      "source-table": null
    },
    "type": "query"
  },
  "display": "table",
  "visualization_settings": {}
}
```
which is exactly [the query I used to generate the hash](https://github.com/metabase/metabase/compare/e2e-deprecate-open-notebook-editor?expand=1#diff-297f82a7472c5d85e6e1448601add09406a8a1359b4c288ddb534911ae4ce28aR22) for this helper function.

## Important!
Just realized that `display` shouldn't be provided to this query because then Metabase loses the ability to intelligently choose the appropriate visualization. Fixed it in https://github.com/metabase/metabase/pull/21552/commits/9454e937effd8bc48c9f61e8c9559d401cbca710.

How to test this?
**UI**
1. Visit `/`
2. Click on "New"
3. Click on "Question"
4. Click on "Sample Database"
5. Open "Orders" table
6. Click on "Pick the metric you want to see"
7. Visualize
8. You should see the scalar value 18,760 rather than a table

**With the new helper function** 
1. `startNewQuestion()`
2. Click on "Sample Database"
3. Open "Orders" table
4. Click on "Pick the metric you want to see"
5. Visualize
6. You should see the scalar value 18,760 rather than a table

![image](https://user-images.githubusercontent.com/31325167/162509179-eea01fd7-c824-4f79-8f21-94df5065ef67.png)
